### ISSUE TITLE
Is it possible to create population with 5+ objectives?

### ISSUE THREAD

**Issue by ChameleonTartu** [msu-coinlab/pymoo#98]
I am so excited to find this project but I cannot find a way to create a population.

My idea is to generate a population with 5 or more different characteristics.
For instance,
objective 1 - age
objective 2 - wages/year
objective 3 - num kids
objective 4 - investments/year
objective 5 - traveling costs/year

Is it possible to do it?

If not, can you possibly recommend me how can I submit PR and adjust the code, so it will be possible?

**Comment by julesy89** [comment]
Yes, you can define the objective regarding features. 

Your description does not clearly say what you are intending to do. Do you have an objective function to optimize?
Or do you just want to extract non-dominated solutions from a set? Or maybe, even more, including some decision making?

**Comment by ChameleonTartu** [comment]
@julesy89 I want to extract non-dominated solutions from the set.

**Comment by julesy89** [comment]
Yes you can do that. This is used frequently in the algorithm implementations in pymoo.

Depending on how many individuals/entries in the data set you have and how many objectives different Non-Dominated Sorting Algorithms will perform better than others. (If only a few entries it will not really matter).

The standard `FastNonDominatedSorting` algorithm proposed in the well-known NSGA2 paper can be executed by:

```
import numpy as np
from pymoo.util.nds.non_dominated_sorting import NonDominatedSorting

F = np.random.random((10000, 5))
nds = NonDominatedSorting(method="fast_non_dominated_sort").do(F, only_non_dominated_front=True)
```
where nds contains the indices of all non-dominatd solutions.

You also might want to look at `efficient_non_dominated_sort ` as shown in the code to below. This is a benchmark of two NDS algorithm on a random data set.

```
from timeit import timeit

import numpy as np

from pymoo.util.nds.non_dominated_sorting import NonDominatedSorting

F = np.random.random((10000, 5))

nds = NonDominatedSorting(method="fast_non_dominated_sort").do(F, only_non_dominated_front=True)
print("Fast Non-Dominated Sorting")
print("result", np.sort(nds))
print("time", timeit("NonDominatedSorting(method='fast_non_dominated_sort').do(F, only_non_dominated_front=True)", number=10, globals=globals()))

nds = NonDominatedSorting(method='efficient_non_dominated_sort').do(F, only_non_dominated_front=True)
print("Fast Non-Dominated Sorting")
print("result", np.sort(nds))
print("time", timeit("NonDominatedSorting(method='efficient_non_dominated_sort').do(F, only_non_dominated_front=True)", number=10, globals=globals()))


```

**Comment by ChameleonTartu** [comment]
@julesy89 Thanks for the help! I am pretty new to genetic algorithms.

I wanted to ask that in NSGA - II, for example, they use three steps:
- Non-dominated sorting
- Crowd-distancing assignment
- Sorting

Do I understand right that in the snippet you provided, there are points (1) and (3) but not (2)?

My idea is to apply NSGA-II, NSGA-III, and PSO to my problem and analyze the performance of this three algos. I hope it makes any sense. It is a bit hard to explain myself in a new domain.

**Comment by julesy89** [comment]
NSGA2 does first NDS and then crowding for each front. And then it truncate the population depending on the population size. 

I am not sure if you just want to do a survival ONCE? Then you are actually not using the algorithms, because they would need an objective function. Correct me if I am wrong. However, you can use the survival anyway. 

PSO was design for single-objective, NSGA2 for bi-objective and NSGA3 for many-objectives problems. You have 5 objectives which means many-objective. Of cource it depends on the context but I do not think a comparisons of these would make sense.

**Comment by ChameleonTartu** [comment]
@julesy89 Yeap, it may sound a bit strange but assumingly that I can modify NSGA2 and PSO algorithms to run against 5-objective. How will the call look like? 

I got into a small weird R&D project, and I have a task to compare PSO, NSGA2, and NSGA3 algos between each other.

Regarding the previous comments, yes, I understand non-dominated sorting will return a subset, and I would need to run algorithms many times until I get the expected population. For example, if I want 10000 individuals, and after the first run, I get only 5, I would run it as many times as needed until I get 10000 individuals. After this, I apply crowd-distancing to remove outliers. Manhattan (crowd-distance) should work on hyper-vectors, so this is not as a big of a deal. I am somewhat concerned about how to make the initial setup and to visualize the result afterward.

P.S. I read the NSGA2 paper, so if I described anything wrong, I am not 100% understand the details yet.
